ddl/index.go

```diff
@@ -1536,17 +1536,56 @@ func (w *worker) updateReorgInfoForPartitions(t table.PartitionedTable, reorg *r
 	return false, errors.Trace(err)
 }
 
-func findIndexesByColName(indexes []*model.IndexInfo, colName string) ([]*model.IndexInfo, []int) {
-	idxInfos := make([]*model.IndexInfo, 0, len(indexes))
-	offsets := make([]int, 0, len(indexes))
-	for _, idxInfo := range indexes {
-		for i, c := range idxInfo.Columns {
-			if strings.EqualFold(colName, c.Name.L) {
-				idxInfos = append(idxInfos, idxInfo)
-				offsets = append(offsets, i)
-				break
+type indexesToChange struct {
+	indexInfo *model.IndexInfo
+	offset    int  // column offset in idxInfo.Columns
+	isTemp    bool // this is a temp index created by a previous modify column job.
```

Comment:
IMHO it's better to add more details to the 'temp' index in the comment. The logic here is not easy to follow.

The descriptions in https://pingcap.feishu.cn/docx/doxcn53f2eGAxYPZPW01drDmLRf?from=from_lark_index_search is clear. How about putting them in the comment?

Reply:
Sure. I have updated it in the comment.